### PR TITLE
feat: [PL-28467]: GCP secret manager

### PR DESCRIPTION
**Pull request by swarajcheguri-harness**
### Summary
image
image
image
image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 49 Issues 
* Code issue 49 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11459)

File: src/modules/35-connectors/components/CreateConnector/GCPSecretManager/CreateGCPSecretManager.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  isEdit: boolean
+  spec: any
+}
+
+const CreateGCPSecretManager: React.FC<CreateConnectorModalProps> = props => {
+  const { onClose, onSuccess } = props
+  const { getString } = useStrings()
+  return (
+    <StepWizard
+      icon={getConnectorIconByType(Connectors.AWS_KMS)}
```
Comment (upadhyay-an) [inline src/modules/35-connectors/components/CreateConnector/GCPSecretManager/CreateGCPSecretManager.tsx:38]: GCP Secret Manager and AWS Key Management are going to have same icon?

File: src/modules/35-connectors/components/CreateConnector/GCPSecretManager/CreateGCPSecretManager.module.scss [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+ */
+
+.wizardNav {
+  grid-template-columns: 360px 1fr !important;
```
Comment (upadhyay-an) [inline src/modules/35-connectors/components/CreateConnector/GCPSecretManager/CreateGCPSecretManager.module.scss:9]: Is this really necessary? The wizard modal window is flexible by itself.

File: src/modules/35-connectors/constants.ts [hunk 3]
```diff
@@ -111,7 +111,8 @@ export const Connectors: ConnectorType = {
   AWSSECRETMANAGER: 'AwsSecretManager',
   JENKINS: 'Jenkins',
   CUSTOM_SECRET_MANAGER: 'CustomSecretManager',
-  ELK: 'ELK'
+  ELK: 'ELK',
+  GcpSM: 'GcpSM'
```
Comment (upadhyay-an) [inline src/modules/35-connectors/constants.ts:115]: Is this like backend defined thing which can't be changed? Or else GDP_SECRET_MANAGER / GCP_SM would be more fitting.

File: src/modules/35-connectors/components/CreateConnector/GCPSecretManager/views/GCPSecretManagerConfig.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            category: Category.CONNECTOR,
+            connector_type: Connectors.GcpSecretManager
+          })
+          nextStep?.({ ...props.connectorInfo, ...prevStepData, ...formData } as StepDetailsProps)
+        }}
+      >
+        {() => {
+          return (
+            <FormikForm>
+              <Container style={{ minHeight: 420, maxWidth: 600 }} margin={{ top: 'medium', bottom: 'xxlarge' }}>
```
Comment (upadhyay-an) [inline src/modules/35-connectors/components/CreateConnector/GCPSecretManager/views/GCPSecretManagerConfig.tsx:98]: The idea is to move away from inline styling.

**Review by upadhyay-an — APPROVED** [review]

**Review by upadhyay-an — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**